### PR TITLE
ENH: preserve non-nano DTA/TDA in Index/Series/DataFrame

### PR DESCRIPTION
**Pull request by jbrockmendel**
Continue to cast non-nano ndarrays, but if a user goes out of there way to create a non-nano DTA/TDA, preserve it in the other constructors.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
cc @datapythonista

**Comment by datapythonista** [comment]
Looks really cool.

Not sure if for the tests I'd compare to the literal of the dtype we expect. We're testing that the type of the array are preserved. This could be tested in general (not sure if it is), but feels a bit strange that this is done in the tests specific to non-nano, instead of making sure the dtype is the non-nano we want. Not important, just sharing, if you like this approach better, is fine with me. I guess we're testing the dtype of the array elsewhere already.

File: pandas/core/internals/construction.py [hunk 1]
```diff
@@ -537,15 +537,16 @@ def treat_as_nested(data) -> bool:
 # ---------------------------------------------------------------------
 
 
-def _prep_ndarray(values, copy: bool = True) -> np.ndarray:
+def _prep_ndarray(
```
Comment (jreback) [inline pandas/core/internals/construction.py:540]: maybe rename to prep_array_like or similar?

**Review by mroeschke — COMMENTED** [review]
Looks fairly good. Just needs a rebase

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**